pkg/azuredisk/nodeserver.go

```diff
@@ -494,15 +568,36 @@ func getNodePublishMountOptions(req *csi.NodePublishVolumeRequest) []string {
 	return mountOptions
 }
 
-func (d *Driver) getBlockSizeBytes(devicePath string) (int64, error) {
-	output, err := d.mounter.Exec.Run("blockdev", "--getsize64", devicePath)
+func statFS(path string) (available, capacity, used, inodesFree, inodes, inodesUsed int64, err error) {
+	statfs := &unix.Statfs_t{}
+	err = unix.Statfs(path, statfs)
 	if err != nil {
-		return -1, fmt.Errorf("error when getting size of block volume at path %s: output: %s, err: %v", devicePath, string(output), err)
+		err = fmt.Errorf("Failed to get fs info on path %s: %v", path, err)
+		return
 	}
-	strOut := strings.TrimSpace(string(output))
-	gotSizeBytes, err := strconv.ParseInt(strOut, 10, 64)
+
+	// Available is blocks available * fragment size
+	available = int64(statfs.Bavail) * int64(statfs.Bsize)
+
+	// Capacity is total block count * fragment size
+	capacity = int64(statfs.Blocks) * int64(statfs.Bsize)
+
+	// Usage is block being used * fragment size (aka block size).
+	used = (int64(statfs.Blocks) - int64(statfs.Bfree)) * int64(statfs.Bsize)
+
+	inodes = int64(statfs.Files)
+	inodesFree = int64(statfs.Ffree)
+	inodesUsed = inodes - inodesFree
+
+	return
+}
+
+func isBlockDevice(fullPath string) (bool, error) {
+	var st unix.Stat_t
+	err := unix.Stat(fullPath, &st)
```

Comment:
could you run below command to make sure this driver would build on windows?
```
make azuredisk-windows
```
We are going to add windows build test

Reply:
`make azuredisk-windows` is failing for this code. I'll start working on the fix.

Reply:
and also there is sanity test failure, pls fix that, it would require a few more efforts on this.